SimG4Core/CustomPhysics/src/CMSQGSPSIMPBuilder.cc

```diff
@@ -0,0 +1,57 @@
+
+#include "SimG4Core/CustomPhysics/interface/CMSQGSPSIMPBuilder.h"
+#include "G4SystemOfUnits.hh"
+#include "G4ParticleDefinition.hh"
+#include "G4ParticleTable.hh"
+#include "G4ProcessManager.hh"
+#include "G4ExcitationHandler.hh"
+
+#include "SimG4Core/CustomPhysics/interface/CMSSIMPInelasticXS.h"
+#include "SimG4Core/CustomPhysics/interface/CMSSIMP.h"
+
+CMSQGSPSIMPBuilder::CMSQGSPSIMPBuilder(G4bool quasiElastic) {
+  theMin = 12 * GeV;
+
+  theModel = new G4TheoFSGenerator("QGSP");
+
+  theStringModel = new G4QGSModel<G4QGSParticipants>;
+  theStringDecay = new G4ExcitedStringDecay(theQGSM = new G4QGSMFragmentation);
+  theStringModel->SetFragmentationModel(theStringDecay);
+
+  theCascade = new G4GeneratorPrecompoundInterface;
+  thePreEquilib = new G4PreCompoundModel(new G4ExcitationHandler);
+  theCascade->SetDeExcitation(thePreEquilib);
+
+  theModel->SetTransport(theCascade);
+  theModel->SetHighEnergyGenerator(theStringModel);
+  if (quasiElastic) {
+    theQuasiElastic = new G4QuasiElasticChannel;
+    theModel->SetQuasiElasticChannel(theQuasiElastic);
+  } else {
+    theQuasiElastic = nullptr;
+  }
+}
+
+CMSQGSPSIMPBuilder::~CMSQGSPSIMPBuilder() {
+  delete theStringDecay;
+  delete theStringModel;
+  delete thePreEquilib;
```

Comment:
@fabiocos , lines 38-39 should be removed (delete thePreEquilib; delete theCascade)